### PR TITLE
Adding swagger changes for routemaps feature

### PR DESCRIPTION
**Pull request by arganapathy**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json [hunk 1]
```diff
@@ -1675,6 +1675,240 @@
...
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. Returns the details of the VirtualHub retrieved.",
+            "schema": {
+              "$ref": "#/definitions/VirtualHub"
```
Comment (zaynchen115) [inline specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json:1720]: Do we need a definition of routemap

File: specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json [hunk 2]
```diff
@@ -5683,6 +5917,126 @@
         }
       }
     },
+    "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualHubs/{virtualHubName}/inboundRoutes": {
```
Comment (zaynchen115) [inline specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json:5920]: For inboundRoutes and outboundRoutes, do we first need implement it in NRP and nfv?

**Comment by raych1** [comment]
> Hi @arganapathy, Your PR has some issues. Please fix the CI sequentially by following the order of `Avocado, semantic validation, model validation, breaking change, lintDiff`. If you have any questions, please post your questions in this channel https://aka.ms/swaggersupport.
> 
> Task	How to fix	Priority
> Avocado	[Fix-Avocado](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/ci-fix.md#avocado)	High
> Semantic validation	[Fix-SemanticValidation-Error](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/ci-fix.md#semantic-validation)	High
> Model validation	[Fix-ModelValidation-Error](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/ci-fix.md#model-validation)	High
> LintDiff	[Fix-LintDiff](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/ci-fix.md#linter-validation)	high
> If you need further help, please feedback via [swagger feedback](https://aka.ms/swaggerfeedback).

@arganapathy , can you fix CI check failures?

File: specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json (hunk 1 above)
Comment (raosuhas) [reply specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json:1720]: Please fix .. GET and PUT need to return the same item

File: specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json (hunk 2 above)
Comment (raosuhas) [inline specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json:5920]: >inboundRoutes [](http://example.com/codeflow?start=128&length=13)

If you want these POST actions to be usable in an ARM template they need to be prefixed with list* 

Examples :[https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#list](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#list)

**Comment by raosuhas** [comment]
Looks mostly fine. Added a couple minor comments

**Comment by dedhar** [comment]
1. Did you verify json by creating route map with rules using examples via postman / resources.azure.com or comparing with jsons from portal? 
2. Can you try put on any one connection to ensure routing configuration with route map is picked up?
3. Are back references getting populated after connection associates to route map?
4. Can you try sample get inbound and outbound route maps?
5. Does unit testing cover above or we need to manually verify jsons?
6. Is List on route maps returning all route maps?
7. Compare with hub route table and get effective routes to ensure we didn't miss anything on hub resource

File: specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json (hunk 2 above)
Comment (dedhar) [reply specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json:5920]: Its already implemented in nrp and nfv

File: specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json [hunk 3]
```diff
@@ -5683,6 +5917,126 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
```
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json:5976]: does this need x-ms-pageable - check get effective routes once

File: specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json (hunk 2 above)
Comment (arganapathy) [reply specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json:5920]: We will resolve this in future api releases.

**Comment by arganapathy** [comment]
@suhasaraos  could you take another look?

**Comment by arganapathy** [comment]
@suhasaraos @raych1 Had a couple questions regarding the ci failures . For the breaking change i  see that the differences with previous api are in files which arent even part of my pr. Could you advise on how to resolve this?
image
Also for the sdk failures i'm not able to parse any useful information from the logs. I've attached one for reference..any idea what the issue is? 
[logs (9).txt](https://github.com/Azure/azure-rest-api-specs/files/9320846/logs.9.txt)

**Comment by raych1** [comment]
@ArcturusZhang @msyyc , can you have a look at GO/Python SDK errors?

**Comment by msyyc** [comment]
Hi @arganapathy Please update `readme.python.md` with [latest code of main branch](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/network/resource-manager/readme.python.md) to fix python failure.

**Comment by arganapathy** [comment]
@msyyc thanks for the help. Now it seems only the powershell sdk is failing. Would you have an idea why? I'm not able to make much sense of the msg from the log
 <li>
    <div>
      <code>❌</code>Failed [<a
        href="https://portal.azure-devex-tools.com/api/sdk-dl-pub?p=Azure/20091/azure-powershell/logs.txt"
        >Logs</a
      >] Generate from 20c31d63fa77dbf5994c41dc20ebbee6c23e655e. SDK Automation
      14.0.0
    </div>
    <pre>
command	sh ./tools/SwaggerCI/init.sh ../azure-powershell_tmp/initInput.json ../azure-powershell_tmp/initOutput.json
command	pwsh ./tools/SwaggerCI/psci.ps1 ../azure-powershell_tmp/generateInput.json ../azure-powershell_tmp/generateOutput.json</pre
    >
  </li>
  <li>
    <div>
      <span
        ><code>❌</code><b>Az.network</b> [<a
          href="https://portal.azure-devex-tools.com/api/sdk-dl-pub?p=Azure/20091/azure-powershell/Az.network/logs.txt"
          >View full logs</a
        >]&nbsp; [<a
          href="https://github.com/azure-sdk/azure-powershell/pull/1550"
          >Preview SDK Changes</a
        >]
      </span>
    </div>
  </li>
</ul>

**Comment by ArcturusZhang** [comment]
I only see additions in this PR but the go SDK breaking change detection says there are a lot of deletions of things.
It is very probable that these breaking changes are not introduced by this PR.
I will let @tadelesh know about this

**Comment by tadelesh** [comment]
Breakings are from this [PR](https://github.com/Azure/azure-rest-api-specs/pull/19985). We will have a final review when merging into main branch.

**Comment by arganapathy** [comment]
It seems ci failures for breaking change and powershell sdk can be ignored. @MikhailTryakhov @suhasrao @msyyc can you approve?

**Comment by dedhar** [comment]
Approving PR from vWAN

**Comment by msyyc** [comment]
approve for python

**Comment by raych1** [comment]
The change identified by `cross version breaking change` validation are not brought in by this PR.

File: specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json (hunk 2 above)
Comment (arganapathy) [reply specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json]: done

Comment (raosuhas) [reply specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/virtualWan.json]: That would be considered to be a breaking change and would have to to go through the whole decommisioning process if you do it later. This is a simple rename so I would suggest to fix it now.

**Review by raosuhas — APPROVED** [review]
:shipit:

**Review by raych1 — APPROVED** [review]

**Pull request merged**